### PR TITLE
Update URL links and Space documentation summaries

### PR DESCRIPTION
**Pull request by mgoulao**
# Description

* Fix some broken links and typos in the codebase.
* Update text on the main page "Gymnasium is a maintained fork of OpenAI’s Gym library. The Gymnasium interface is simple, pythonic, and capable of representing general RL problems:"
* Update `Spaces` structure

### CODE REVIEW

File: docs/content/environment_creation.md [hunk 1]
```diff
@@ -307,7 +307,7 @@ register(
 The environment ID consists of three components, two of which are optional: an optional namespace (here: `gym_examples`), a mandatory name (here: `GridWorld`) and an optional but recommended version (here: v0). It might have also been registered as `GridWorld-v0` (the recommended approach), `GridWorld` or `gym_examples/GridWorld`, and the appropriate ID should then be used during environment creation.
 
 The keyword argument `max_episode_steps=300` will ensure that GridWorld environments that are instantiated via `gymnasium.make`
-will be wrapped in a `TimeLimit` wrapper (see [the wrapper documentation](https://www.gymlibrary.dev/pages/wrappers/index) 
+will be wrapped in a `TimeLimit` wrapper (see [the wrapper documentation](/api/wrappers)
```
Comment (pseudo-rnd-thoughts) [inline docs/content/environment_creation.md:310]: Should this not start with `www.gymnasium.farama...`

File: docs/content/environment_creation.md [hunk 2]
```diff
@@ -372,7 +372,7 @@ also perfectly fine (but remember to add  wrappers as well!).
 Oftentimes, we want to use different variants of a custom environment, or we want to
 modify the behavior of an environment that is provided by Gymnasium or some other party. 
 Wrappers allow us to do this without changing the environment implementation or adding any boilerplate code.
-Check out the [wrapper documentation](https://www.gymlibrary.dev/content/wrappers/) for details on how to 
+Check out the [wrapper documentation](/content/wrappers/) for details on how to
```
Comment (pseudo-rnd-thoughts) [inline docs/content/environment_creation.md:375]: This is a different link to the one above

File: docs/content/environment_creation.md [hunk 3]
```diff
@@ -389,7 +389,7 @@ print(wrapped_env.reset())     # E.g.  [3 0 3 3], {}
 
 Wrappers have the big advantage that they make environments highly modular. For instance, instead of flattening the 
 observations from GridWorld, you might only want to look at the relative position of the target and the agent. 
-In the section on [ObservationWrappers](https://www.gymlibrary.dev/content/wrappers/#observationwrapper) we have implemented
+In the section on [ObservationWrappers](/content/wrappers/#observationwrapper) we have implemented
```
Comment (pseudo-rnd-thoughts) [inline docs/content/environment_creation.md:392]: Same comment as above

File: docs/content/environment_creation.md (hunk 1 above)
Comment (mgoulao) [reply docs/content/environment_creation.md:310]: It can start, but I think it's better to use relative paths, this way we don't need to update domains.

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Pull request merged**